docs/docsite/rst/porting_guides/porting_guide_2.10.rst

```diff
@@ -92,6 +92,7 @@ The following modules will be removed in Ansible 2.14. Please update your playbo
 Noteworthy module changes
 -------------------------
 
+* Ansible modules created with ``add_file_common_args=True`` added a number of undocumented arguments which were mostly there to ease implementing certain action plugins. The undocumented arguments ``src``, ``follow``, ``force``, ``content``, ``backup``, ``remote_src``, ``regexp``, ``delimiter``, and ``directory_mode`` are now no longer added. Modules relying on these options to be added need to specify them by themselves.
```

Comment:
Is there any way a module owner would know they used that ``add_file_common_args=True`` ?  And what happens to a user of one of those modules if the module owner doesn't make a code change in 2.10 to match this?

Reply:
Module owners explicitly have to set `add_file_common_args=True`, so they should know. For most modules using `add_file_common_args=True`, this should have absolutely no effect, since these parameters weren't documented and the sanity checks ignore them explicitly (i.e. essentially nobody knows that they are undocumented, and thus that they exist at all). It can happen in a few cases (like the ones I fixed in this PR) that modules forgot to declare some of these options by themselves, because "it just worked".

There are 29 modules in ansible/ansible which use this flag. These ones we can check manually (a lot of work, but not impossible), but if someone has a custom module which depends on one of these options, we can't really avoid this. But for this similar arguments as for the modules in ansible/ansible apply: this shouldn't affect most modules using `add_file_common_args=True`, and a huge majority of modules doesn't even use that.

Reply:
ok thanks for the explanation!